enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,294 @@
+
+### User Stories
+
+* As an Openshift user, I would like to be able to run my container as root without needing to be trusted by the platform.
+* As a user of Openshift Devspaces, I would like to run podman within the Devspace.
+* As an Openshift admin, I would like to run untrusted users on a tighter security profile than the SCC restricted-v2.
+* As an Openshift admin, I would like to ensure pods that request user namespaces are confined to one.
+
+### Goals
+
+- Enable Openshift users to request a pod be put in a user namespace
+- Update SCC to take user namespaces into account when choosing the security profile of a container
+- Add support for users to run podman within an Openshift pod without being privileged.
+
+### Non-Goals
+
+- 
+
+## Proposal
+
+There are three pieces to this proposal:
+- Extend SCC to be aware of the hostUsers field:
+    - Add a new field AllowHostUser to SCC, which will be Allowed by default
+    - Add a new SCC to the default list: restricted-v3
+        - This SCC will be identical to restricted-v2, but have AllowHostUsers set to Disallowed
+    - Add a new SCC to the default list: container-in-pod. It will have:
+        - SELinux context set to MustRunAs.Type: container_engine_t
+        - RunAsUserStrategy: RunAsAny
+        - RequiredDropCapabilities: None
+        - AllowHostUser: Disallowed
+        - And otherwise mirror the restricted-v2 profile
+- Add the features UserNamespacesSupport and UserNamespacesPodSecurityStandards to the list that qualify a cluster as TechPreviewNoUpgrade
+
+### Workflow Description
+
+N/A
+
+### API Extensions
+
+- Add AllowHostUser to SCC. This relies on approval from the apiserver team.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+- Support for a user to change the node config object will have to be investigated.
+
+#### Standalone Clusters
+
+I don't think there are any special topology considerations for standalone.
+
+#### Single-node Deployments or MicroShift
+
+From my understanding, there should not be any large resource consumption changes for this feature.
+
+### Implementation Details/Notes/Constraints
+
+#### SecurityContextConstraint Updates
+
+##### `AllowHostUsers` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // hostUsers determines if the policy allows host users in containers.
+       // Valid values are "Allowed", "Disallowed" and omitted.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default.
+       // The current default is "Allowed".
+       // +openshift:enable:FeatureGate=UserNamespacesSupport
+       // +kubebuilder:validation:Enum="Allowed";"Disallowed";""
+       // +kubebuilder:validation:Optional
+       HostUsers HostUsersStrategyType `json:"hostUsers" protobuf:"varint,26,opt,name=allowHostUsers"`
```

Comment:
Should this be `AllowHostUsers`? varint does not seem correct either. As does the "hostUsers" and "allowHostUsers" mismatch.

Reply:
good points, I must have done some weird copying

Reply:
Thanks!

Reply:
The version there was suggested by me. We are discouraging the use of new booleans in APIs and, within OpenShift, even forbidding it. It's preferable to use enums since they can be expanded over time, where currently, these bool values will never be able to be expanded. The version here would adhere to current API guidelines, even though it creates a new pattern